util/network/src/connection.rs

```diff
@@ -158,8 +158,8 @@ impl<Socket: GenericSocket> GenericConnection<Socket> {
 			}
 			if self.send_queue.is_empty() {
 				self.interest.remove(Ready::writable());
-				io.update_registration(self.token)?;
 			}
+			io.update_registration(self.token)?;
```

Comment:
Not too clear on the mio specifics, but does it cause any issues to `reregister` interest multiple times before a `timeout` or `notify` event is handled? I thought that was the original purpose of waiting until the send queue was empty.